### PR TITLE
Fix/apply inference

### PR DESCRIPTION
**Pull request by MksmOrlov**

### CODE REVIEW

**Review by AlexandrZagorskiy — APPROVED** [review]

File: scl/cxx/inferenceModule/logic/LogicExpression.cpp [hunk 1]
```diff
@@ -253,26 +251,27 @@ LogicFormulaResult ImplicationExpressionNode::compute() const
 
   if (!isLeftGenerated)
   {
-    SC_LOG_DEBUG("*** Left part shouldn't be generated");
-    leftResult = operands[0]->compute();
-    rightResult = (isRightGenerated ? rightAtom->generate(leftResult.replacements) : operands[1]->compute());
+    SC_LOG_DEBUG("If part shouldn't be generated");
```
[the remark this answers is not in the document]
Comment (MksmOrlov) [reply scl/cxx/inferenceModule/logic/LogicExpression.cpp]: Thanx

File: scl/cxx/inferenceModule/test/testStructures/LogicModule/trueComplexImplicationRule.scs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+rules_set
+    -> rrel_1: { lr_complex };;
+
+inference_logic_test_question
+    <- action_direct_inference;
+    -> rrel_1: target_template;
+    -> rrel_2: rules_set;
+    -> rrel_3: input_structure;
+    -> rrel_4: output_structure;;
```
Comment (SHtress) [inline scl/cxx/inferenceModule/test/testStructures/LogicModule/trueComplexImplicationRule.scs:101]: Newline

**Review by SHtress — APPROVED** [review]

File: scl/cxx/inferenceModule/test/testStructures/LogicModule/trueComplexImplicationRule.scs (hunk 2 above)
Comment (MksmOrlov) [reply scl/cxx/inferenceModule/test/testStructures/LogicModule/trueComplexImplicationRule.scs]: исправил

File: scl/cxx/inferenceModule/logic/LogicExpression.cpp [hunk 3]
```diff
@@ -214,7 +214,8 @@ LogicExpressionResult NotExpressionNode::check(ScTemplateParams params) const
 LogicFormulaResult NotExpressionNode::compute(LogicFormulaResult & result) const
 {
   const LogicFormulaResult & formulaResult = operands[0]->compute(result);
-  SC_LOG_DEBUG("Sub formula in negation returned " + to_string(formulaResult.value));
+  std::string formulaValue = (formulaResult.value ? "true" : "false");
+  SC_LOG_DEBUG("Sub formula in negation returned " + formulaValue);
```
Comment (AlexandrZagorskiy) [inline scl/cxx/inferenceModule/logic/LogicExpression.cpp:218]: Why log says about negation? Is this method used for every logic operation?

**Review by AlexandrZagorskiy — APPROVED** [review]

File: scl/cxx/inferenceModule/logic/LogicExpression.cpp (hunk 3 above)
Comment (MksmOrlov) [reply scl/cxx/inferenceModule/logic/LogicExpression.cpp:218]: NotExpressionNode::compute

File: scl/cxx/inferenceModule/test/units/TestComplexFormulas.cpp [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  context.CreateEdge(ScType::EdgeAccessConstPosPerm, InferenceKeynodes::action_direct_inference, test);
+  EXPECT_TRUE(utils::AgentUtils::applyAction(&context, test, WAIT_TIME));
+  EXPECT_TRUE(context.HelperCheckEdge(
+      scAgentsCommon::CoreKeynodes::question_finished_unsuccessfully, test, ScType::EdgeAccessConstPosPerm));
+
+  shutdown();
+  context.Destroy();
+}
+
+TEST_F(InferenceComplexFormulasTest, EquivalencesNested)
```
[the remark this answers is not in the document]
Comment (MksmOrlov) [reply scl/cxx/inferenceModule/test/units/TestComplexFormulas.cpp:269]: There is a weired thing in equivallence compute method, I didn't check it. Also there is test structures from other project (nrel_equipment_usage_points, nrel_connectivity_nodes and so on) which is not pretty for scl-machine. So, test is not commented and will be changed later

File: scl/cxx/inferenceModule/logic/LogicExpression.cpp (hunk 3 above)
Comment (AlexandrZagorskiy) [reply scl/cxx/inferenceModule/logic/LogicExpression.cpp:218]: Should it be renamed to negation? Not obvious for me, but maybe there is another context, and usage of such concept is incorrect here

File: scl/cxx/inferenceModule/test/units/TestComplexFormulas.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexandrZagorskiy) [reply scl/cxx/inferenceModule/test/units/TestComplexFormulas.cpp:269]: Got it, thanks

File: scl/cxx/inferenceModule/logic/LogicExpression.cpp (hunk 3 above)
Comment (MksmOrlov) [reply scl/cxx/inferenceModule/logic/LogicExpression.cpp:218]: Yes, we have task to refactor it. Such as "rule", "statement", "or", "and"

**Review by SHtress — APPROVED** [review]

**Pull request merged**